src/ui.py

```diff
@@ -50,34 +52,33 @@ def on_confirm_discard_response(_dialog, response, self):
 
 
 def on_add_data_response(dialog, response, self, import_settings=None):
-    try:
+    with contextlib.suppress(GLib.GError):
         files = dialog.open_multiple_finish(response)
         file_import.import_from_files(self, files, import_settings)
-    except GLib.GError:
-        pass
 
 
 def on_export_data_response(dialog, response, self, multiple):
-    try:
+    with contextlib.suppress(GLib.GError):
         if multiple:
-            path = dialog.select_folder_finish(response).peek_path()
+            directory = dialog.select_folder_finish(response)
+            for item in self.datadict.values():
+                file = directory.get_child_for_display_name(f"{item.name}.txt")
+                file_io.save_file(file, item.xdata, item.ydata)
         else:
-            path = dialog.save_finish(response).peek_path()
-        file_io.save_file(self, path)
-    except GLib.GError:
-        pass
+            file = dialog.save_finish(response)
+            # return value is not subscriptable, so we loop over one item
```

Comment:
Not sure which is better, but in principle you can convert the values object to a list with list(self.datadict.values), which is iterable.

The loop can then be replaced by a single item = list(self.datadict.values())[0]

Reply:
Ah, i trie `item = set(self.datadict.values())[0]` and still got the non-subscriptable error. This is probably a better solution, as the datadict *might* change between invoking the dialog and accepting it.